covid_graphs/covid_graphs/country_graph.py

```diff
@@ -56,6 +56,26 @@ def _get_display_range(
     return start_date, display_until
 
 
+# TODO: move this to another file
+def _get_earliest_displayable_idx(
+    fitted_formulas: Iterable[FittedFormula], max_peak_distance: datetime.timedelta
+) -> int:
```

Comment:
What is the intended logic here? Can you maybe add a comment? (Then as a reviewer I can check that the code does what is intended.)

Reply:
Is it the longest interval ending today where `max_peak - min_peak <= const`? Then maybe `while` is more suitable, sth like:
```py
start_idx = len(peak_days) - 1
max_peak, min_peak = peak_days[-1], peak_days[-1]
while max_peak - min_peak < const and start_idx > 0:
    start_idx -= 1
    max_peak = max(max_peak, peak_days[start_idx])
    min_peak = min(min_peak, peak_days[start_idx])
return start_idx + 1
```

Reply:
Also, I think a more simple condition that the peak is within some distance from the last prediction would be good enough for our purposes.

Reply:
Added a comment and renamed the function. I thought about your suggested simplification but decided that this one seems better suited for what we want, e.g. in the case of a slider. But I don't have a strong opinion on this one.

I disagree with your code suggestion, it has many issues, e.g. it has many possible off-by-one errors (perhaps not now, but in the future): you return `start_idx+1`, which means you probably chose the wrong invariant. You are also indexing into the array.

I think I might have started with something like what you have but decided on rewriting it.

Reply:
The invariant is simple, `start_idx` is the first index not to be included. Maybe a better name can be chosen. I find it tricky to name variables that get updated, since `start_idx` does not have that property until the loop is done. (The same goes also for the original code.) I also like this version since it does not require reversing the array. But your choice.